base/iterators.jl

```diff
@@ -55,6 +55,18 @@ julia> for (index, value) in enumerate(a)
 """
 enumerate(iter) = Enumerate(iter)
 
+"""
+    enumerate(iter, first_index)
+    enumerate(iter, first_index, last_index)
+    enumerate(iter, range)
+
+Enumerate a subset of an iterator.
+"""
+enumerate(iter, first::Integer) = enumerate(rest(iter, first))
```

Comment:
`rest(iter, first)` only works if the state of `iter` equals the index. You probably need to use drop. Try e.g. `enumerate(filter(isodd, 1:10), ...)` for an example where using `rest` will fail.